iceoryx_examples/ice_multi_publisher/README.md

```diff
@@ -0,0 +1,262 @@
+a subscriber in each of the other applications. The two publishers publish publish two independent counters. The data of bboth is made distinct by using an id, which has the values 1 and 2 respectively. 
+## Expected output
+
+The counters can differ depending on startup of the applications.
+
+### RouDi application
+
+    Reserving 84783040 bytes in the shared memory [/iceoryx_mgmt]
+    [ Reserving shared memory successful ] 
+    Reserving 149655680 bytes in the shared memory [/apex]
+    [ Reserving shared memory successful ] 
+    RouDi is ready for clients
+
+### Publisher application
+
+    Counter Instance sending: id 1 counter 0
+    Counter Instance sending: id 2 counter 0
+    Counter Instance sending: id 1 counter 1
+    Counter Instance sending: id 2 counter 1
+    Counter Instance sending: id 1 counter 2
+    Counter Instance sending: id 1 counter 3
+    Counter Instance sending: id 2 counter 2
+    Counter Instance sending: id 1 counter 4
+    Counter Instance sending: id 1 counter 5
+    Counter Instance sending: id 2 counter 3
+    Counter Instance sending: id 1 counter 6
+### Subscriber application
+
+    Waiting for data ... 
+    Received: id 1 counter 1
+    Received: id 2 counter 1
+    Received: id 1 counter 2
+    Waiting for data ... 
+    Received: id 1 counter 3
+    Received: id 2 counter 2
+    Received: id 1 counter 4
+    Waiting for data ... 
+    Received: id 1 counter 5
+    Received: id 2 counter 3
+    Received: id 1 counter 6
+
+The subscriber application wakes up periodically, looks for data, and if it received any displays the data. It can be seen that that the data arrives in order for any publisher, indciated by a monotonic counter per id. The order of data of different id is indeterminate since the publishers are sending concurrently.
+
+### Resubscriber application
+
+    Waiting for data ... 
+    Unsubscribed ... Subscribe in 3 seconds
+    Subscribe with max number of samples 4
+    Received: id 1 counter 23
+    Received: id 2 counter 12
+    Received: id 1 counter 24
+    Waiting for data ... 
+    Unsubscribed ... Subscribe in 3 seconds
+    Subscribe with max number of samples 1
+    Received: id 1 counter 32
+    Waiting for data ... 
+    Unsubscribed ... Subscribe in 3 seconds
+    Subscribe with max number of samples 2
+    Received: id 2 counter 20
+    Received: id 1 counter 40
+    Waiting for data ... 
+
+The resubscriber application periodically unsubscribes and then subscribes again with a different number of samples which the subscriber can hold. It can be seen that no more than this number of samples are received before unsubscribing again. During the time in which it is unsubscribed data is lost, but when it resubscribes there may already be data available which is then displayed. Due to asynchronous sending and subscribing, it cannot be guaranteed that data is available immediately on resubscribtion and the data can be new (after resubscribe) or stale (before unsubscribe).  
```

Comment:
Please explian why no more than this samples are received. I would say because you only check for samples once and you configured the queue to only store the configured number of latest samples. BTW. When the queue is full and a new sample is published while popping the queue, you can also get more I guess. I'm not sure if we can really fulfiill the contract you describe here. Maybe just focus on the queue capacity and the overflow behavior and don't talk about number of samples that are received

Reply:
It is because the queue size is limited, and you will never see more samples than this queue size allows.

I do not think you can ever get more with the MPMC queue, so I am pretty sure the contract can be fulfilled.
What happens is the following: when we unsubscribe there may be samples left in the queue. During the time we are unsubscribed, nothing is internally received. Once we subscribe again, the queue size changes and potentially the least recent contained samples are ejected. Now the queue size has the capacity as requested and we are able to receive new samples, which uses the usual overflow behavior. This ensures we will never have more samples than this new capacity.

Therefore I do think the claims of this section hold and will only add a bit of explanation regarding the queue size.